Doc/extending/extending.rst

```diff
@@ -127,13 +127,11 @@ Intermezzo: Errors and Exceptions
 
 An important convention throughout the Python interpreter is the following: when
 a function fails, it should set an exception condition and return an error value
-(usually a ``NULL`` pointer).  Exceptions are stored in a static global variable
-inside the interpreter; if this variable is ``NULL`` no exception has occurred.  A
-second global variable stores the "associated value" of the exception (the
-second argument to :keyword:`raise`).  A third variable contains the stack
-traceback in case the error originated in Python code.  These three variables
-are the C equivalents of the result in Python of :meth:`sys.exc_info` (see the
-section on module :mod:`sys` in the Python Library Reference).  It is important
+(usually -1 or a ``NULL`` pointer).  Exception information is stored in
+three members of the interpreter's thread state.  These are ``NULL`` if
+there is no exception.  Otherwise they are the C equivalents of the members
```

Comment:
Actually the second can be NULL or None and the third can be NULL in case of non-normalized exception.

The first is the exception type. NULL if no exception.

The second is either the instance of exception, or argument(s) which will be passed to exception constructor on normalization. NULL or None means no arguments. Tuple means several arguments. Otherwise it is a single argument.

The third is a traceback. Can be NULL for non-normalized exceptions.

Reply:
See also `PyErr_Fetch`, `PyErr_Restore` and `PyErr_NormalizeException`. Terminology should be consistent.

Reply:
Are you saying that a Python user of sys.exc_info might see these other value, or that the existing claim of C equivalents is incomplete?  I presume the latter.  If so, how about this, instead of `These are...`.

If there is no exception, these are ``NULL``.  For a normalized exception , these are the C equivalents of the members of the Python tuple returned by :meth:`sys.exc_info`: the exception type, exception instance, and a traceback object.  For a non-normalized exception, the second will be an argument or tuple of arguments that will be passed to the exception on normalization, or ``NULL`` or None, meaning "no arguments".  The third argument may then be ``NULL``.

Reply:
This seems resolved now.